backend/src/main/java/de/bund/digitalservice/ris/caselaw/adapter/database/jpa/PostgresDocumentationUnitRepositoryImpl.java

```
@@ -683,7 +688,18 @@ private boolean userChanged(
   private DocumentationUnitProcessStepDTO createAndSaveNewProcessStep(
       DocumentationUnitDTO documentationUnitDTO,
       ProcessStepDTO processStepDTO,
-      DocumentationUnitProcessStep currentDocunitProcessStepFromFrontend) {
+      DocumentationUnitProcessStep currentDocunitProcessStepFromFrontend,
+      List<DocumentationUnitProcessStep> docunitProcessStepsFromFrontend) {
+
+    // when assigning a new doc office, all previous process steps are removed
+    boolean isOwnerChangeScenario =
+        (docunitProcessStepsFromFrontend != null
+            && docunitProcessStepsFromFrontend.isEmpty()
+            && "Neu".equals(processStepDTO.getName()));
```

1. Could be extract for a static method, good for direct unit testing later
2. If we cahnge the name of the process step this logic would silently fail. I thought maybe we use the constnat process step id we set on flyway script for mapping. As we use it in both doc service and the process step maybe we could extract it to a static variable and reuse.